src/rust/ensogl/src/display/shape/compound/port.rs

```diff
@@ -0,0 +1,144 @@
+//! This module defines the shapes required for drawing node ports.
+
+use crate::data::color::*;
+use crate::display::shape::*;
+use crate::display::shape::primitive::def::modifier::immutable::*;
+use crate::display::shape::primitive::def::primitive::*;
+use crate::prelude::*;
+
+use crate::display::shape::primitive::def::class::ShapeOps;
+use crate::math::circle_segment::CircleSegment;
+use crate::math::topology::unit::AngleOps;
+use crate::math::topology::unit::Distance;
+use crate::math::topology::unit::PixelDistance;
+use crate::math::topology::unit::Pixels;
+use crate::math::topology::unit::{Angle, Degrees};
+
+use nalgebra::Vector2;
+
+
+
+// ===========================
+// === Port Specification ===
+// ===========================
+
+/// Indicates whether a port is incoming or outgoing.
+#[derive(Clone,Copy,Debug,Eq,PartialEq)]
+pub enum PortDirection {
+    /// Indicates port facing towards the center of its inner circle.
+    Inwards,
+    /// Indicates port facing away from the center of its inner circle.
+    Outwards,
+}
```

Comment:
In case where the names of enums are just 100% clear about what it is about, we are often putting `#[allow(missing_docs)]` on the enum declaration. In such cases docs make the code less readable and do not add anything important to it.

Moreover, why `Inwards` / `Outwards` instead of `In` and `Out` ?